plugins/BEdita/Core/src/Model/Action/SignupUserAction.php

```diff
@@ -174,17 +192,89 @@ protected function createUser(array $data)
         }
         unset($data['status']);
 
+        if (empty($data['auth_provider'])) {
+            return $this->createUserEntity($data, $status, 'signup');
+        }
+
+        $authProvider = $this->checkExternalAuth($data);
+
+        $user = $this->createUserEntity($data, 'on', 'signupExternal');
+
+        // create `ExternalAuth` entry
+        $this->Users->dispatchEvent('Auth.externalAuth', [
+            'authProvider' => $authProvider,
+            'providerUsername' => $data['provider_username'],
+            'userId' => $user->get('id'),
+            'params' => empty($data['provider_userdata']) ? null : $data['provider_userdata'],
+        ]);
+
+        return $user;
+    }
+
+    /**
+     * Create User model entity.
+     *
+     * @param array $data The signup data
+     * @param string $status User `status`, `on` or `draft`
+     * @param string $validate Validation options to use
+     * @return @return \BEdita\Core\Model\Entity\User The User entity created
+     */
+    protected function createUserEntity(array $data, $status, $validate)
+    {
         $action = new SaveEntityAction(['table' => $this->Users]);
 
         return $action([
             'entity' => $this->Users->newEntity()->set('status', $status),
             'data' => $data,
             'entityOptions' => [
-                'validate' => 'signup',
+                'validate' => $validate,
             ],
         ]);
     }
 
+    /**
+     * Check external auth data validity
+     *
+     * To perform external auth check these fields are mandatory:
+     *  - "auth_provider": provider name like "google", "facebook"... must be in `auth_providers`
+     *  - "provider_username": id or username of the provider
+     *  - "access_token": token returned by provider to use in check
+     *
+     * @param array $data The signup data
+     * @return \BEdita\Core\Model\Entity\AuthProvider AuthProvider entity
+     * @throws \Cake\Network\Exception\UnauthorizedException Upon external authorization check failure.
+     */
+    protected function checkExternalAuth(array $data)
+    {
+        /** @var \BEdita\Core\Model\Entity\AuthProvider $authProvider */
+        $authProvider = TableRegistry::get('AuthProviders')->find('enabled', ['name' => $data['auth_provider']])->first();
```

Comment:
Passing options directly to the finder works, but I personally don't like that.

I'd prefer:
```php
$authProvider = TableRegistry::get('AuthProviders')->find('enabled')
    ->where(['name' => $data['auth_provider']])
    ->first();
```